manuscript/template.tex

```diff
@@ -0,0 +1,672 @@
+	dependencies  & dictionary & string                & Per   NPM/Yarn. Key = dependency name. Value = Semantic versioning version string.                                                                                          \\ \hline
+	components    & dictionary & Component             & Listing   of sub-components directly owned by this component. Key = sub-component's   name. Value = sub-component's data structure.                                         \\ \hline
+	parts         & dictionary & Component   List Item & Listing   of the component's parts (and substitutions) defined as sub-components. Key =   part's id. Value = part's key data.                                               \\ \hline
+	functions     & list       & Function              & Listing   of component functions.                                                                                                                                           \\ \hline
+	tools         & dictionary & Component   List Item & Listing   of the required tools (and substitutions) defined as sub-components. Key =   tool's id. Value = tool's key data.                                                  \\ \hline
+	precautions   & list       & string                & Listing   of caution statements (e.g. safety warnings) for the component.                                                                                                   \\ \hline
+	assemblySteps & list       & Activity   Step       & Sequence   of steps required to assemble the component. \\
+	\bottomrule
+	\end{tabularx}
+	\end{adjustwidth}
+\end{table}
+
+The initial goal is to list the components of the SeaLion CubeSat and to generate assembly steps for them.  Through which the architecture can provide detailed documentation to assemble the SeaLion CubeSat.  These assembly instructions would also be generated through the SeaLion mission architecture repository much akin to other documents for SeaLion.  Thus, it creates an easily human readable document from the YAML files code as per the docs-as-code approach.
+
+Eventually, the purpose of all these component data structures is to also create an N2 diagram.  An N2 diagram is used to “is used to capture the interfaces, mechanical and electrical, for all components of the satellite obtained through the mapping process” \cite{asundi13_cubes}.  An example has been provided in figure \ref{fig:N2}.  The end goal is that the architecture would use interfaces and junctions within the YAML files code to automatically generate an N2 diagram.  Thus, it allows for continuous updating that ensures that any changes affecting dependencies within the mission architecture are kept in sync.  This would allow for a team to easily identify “areas where conflicts could arise in interfaces, and highlights input and output dependency assumptions and requirements” \cite{asundi13_cubes}.  Thus, leading to higher efficacy in planning the development and assembly of the satellite.
```

Comment:
Should add note here that "extensions to the DOF data model are under development for enabling such use cases for cubesat systems engineering [footnote to DOF-Cubesat repo]"

Reply:
As to address https://github.com/ODU-CGA-CubeSat/sealion-mission-architecture/pull/255#issuecomment-1558133488 and https://github.com/ODU-CGA-CubeSat/sealion-mission-architecture/pull/255#issuecomment-1558133662, can follow up with "this extended data model would serve as a self-describing specification that new cubesat projects could adopt for their own cubesat missions, whereby the SeaLion mission would serve as the first reference implementation. Additional, training material would be developed around this new data model in effort to help guide incoming cubesat developers.

Reply:
Not sure if this line is the place to address https://github.com/ODU-CGA-CubeSat/sealion-mission-architecture/pull/255#issuecomment-1558133812, but the launch delay spawned the discussion of the battery tradestudy which spawned a tradespace exploration into alternative ConOps, which warranted a new data model for describing a conops https://github.com/ODU-CGA-CubeSat/dof-cubesat/issues/8, which a draft implementation was explored in a test branch as an experimental YAML model https://github.com/ODU-CGA-CubeSat/sealion-mission-architecture/tree/conops-and-power-budget-analysis. Could mention that because this YAML model is a docs-as-code format, the experimental tradespace persists digitally in the sealion-mission-architecture repo

Reply:
Resolved 7ec031a634be042ca2ee49bc96cfc08f6716ac4b